sys/linux/bpf_lsm.txt

```diff
@@ -0,0 +1,73 @@
+# Copyright 2020 syzkaller project authors. All rights reserved.
+# Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+include <uapi/linux/bpf.h>
+
+resource fd_bpf_lsm_prog[fd]
+resource btf_id[int32]
+
+bpf_lsm_func_names = "bpf_lsm_path_mkdir"
+
+syz_btf_id_by_name(name ptr[in, string[bpf_lsm_func_names]]) btf_id
+
+type bpf_insn_tmpl[CODE, DST_REG, SRC_REG, OFF, IMM] {
+	code	const[CODE, int8]
+	dst	const[DST_REG, int8:4]
+	src	const[SRC_REG, int8:4]
+	off	const[OFF, int16]
+	imm	const[IMM, int32]
+}
+
+# Test program 0:
+# int BPF_PROG(test, ..., int ret)
+# {
+#	return 0;
+# }
+# 0: b7 00 00 00 00 00 00 00 r0 = 0
+# 1: 95 00 00 00 00 00 00 00 exit
+
+define insn_cnt_test0	2
+define code_test0_insn0	BPF_ALU64 | BPF_K | BPF_MOV
+define code_test0_insn1	BPF_JMP | BPF_K | BPF_EXIT
+
+insns_test0 {
+	insn0	bpf_insn_tmpl[code_test0_insn0, BPF_REG_0, 0, 0, 0]
+	insn1	bpf_insn_tmpl[code_test0_insn1, 0, 0, 0, 0]
+}
+
+type bpf_attr_bpf_lsm_test_tmpl[INSN_CNT, INSNS] {
+	prog_type		const[BPF_PROG_TYPE_LSM, int32]
+	insn_cnt		const[INSN_CNT, int32]
+	insns			ptr64[in, INSNS]
+	license			ptr64[in, string["GPL"]]
+	log_level		const[0, int32]
+	log_size		const[0, int32]
+	log_buf			const[0, int64]
+	kern_version		const[0, int32]
+	prog_flags		const[0, int32]
+	prog_name		string["test", BPF_OBJ_NAME_LEN]
+	prog_ifindex		const[0, int32]
+	expected_attach_type	const[BPF_LSM_MAC, int32]
+	prog_btf_fd		const[0, int32]
+	func_info_rec_size	const[0, int32]
+	func_info		const[0, int64]
+	func_info_cnt		const[0, int32]
+	line_info_rec_size	const[0, int32]
+	line_info		const[0, int64]
+	line_info_cnt		const[0, int32]
+	attach_btf_id		btf_id
+	attach_prog_fd		const[0, int32]
+} [align[8]]
+
+bpf_attr_bpf_lsm_tests [
+	test0	bpf_attr_bpf_lsm_test_tmpl[insn_cnt_test0, insns_test0]
+]
+
+bpf$BPF_LSM_PROG_LOAD(cmd const[BPF_PROG_LOAD], arg ptr[in, bpf_attr_bpf_lsm_tests], size len[arg]) fd_bpf_lsm_prog
+
+bpf_attr_bpf_lsm_raw_tracepoint_open {
+	name	const[0, int64]
+	prog_fd	fd_bpf_lsm_prog
+}
+
+bpf$BPF_LSM_RAW_TRACEPOINT_OPEN(cmd const[BPF_RAW_TRACEPOINT_OPEN], arg ptr[in, bpf_attr_bpf_lsm_raw_tracepoint_open], size len[arg]) fd_rawtp
```

Comment:
Revert order of declarations in this file to the opposite. This is not C, we are not forced to dump the least important things onto user first :)
Starting from more important things makes code easier to read and follow.